File: app/src/main/java/com/likhit/polis/ui/home/HomeActivity.java
package com.likhit.polis.ui.home;

import android.os.Bundle;

import com.likhit.polis.R;
import com.likhit.polis.base.BaseActivity;
import com.likhit.polis.ui.q_a.QandAFragment;
import com.likhit.polis.ui.recommendations.RecommendationsFragment;
import com.likhit.polis.ui.userdetails.UserDetailsFragment;

import java.util.ArrayList;
import java.util.List;

public class HomeActivity extends BaseActivity implements HomeFragementListener {

    private static final String TAG = "HomeActivity";

    private List<String> answers = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);

        setupToolbar(getString(R.string.app_name), true);

        replaceFragment(HomeFragment.newInstance(), HomeFragment.TAG, false);
    }

    @Override
    public void launchHome() {
        replaceFragment(HomeFragment.newInstance(), HomeFragment.TAG, false);
    }

    @Override
    public void launchUserDetails() {
        replaceFragment(UserDetailsFragment.newInstance(), UserDetailsFragment.TAG, false);
    }

    @Override
    public void launchQA() {
        replaceFragment(QandAFragment.newInstance(), QandAFragment.TAG, false);
    }

    @Override
    public void launchRecommendations() {
        if (answers != null && !answers.isEmpty()) {
            String[] ans = {answers.get(0), answers.get(1), answers.get(2), answers.get(3), answers.get(4)};
            replaceFragment(RecommendationsFragment.newInstance(ans), RecommendationsFragment.TAG, false);
        } else {
            String[] ans = null;
            replaceFragment(RecommendationsFragment.newInstance(ans), RecommendationsFragment.TAG, false);
        }
    }

    public void updateAnswers(List<String> ansers) {

        if (answers == null) {
            answers = new ArrayList<>();
        }
        answers = ansers;
    }
}

File: app/src/main/java/com/likhit/polis/ui/home/HomeActivity.kt
package com.likhit.polis.ui.home

import android.os.Bundle

import com.likhit.polis.R
import com.likhit.polis.base.BaseActivity
import com.likhit.polis.data.models.Policy
import com.likhit.polis.ui.detailPolicy.DetailPolicyFragment
import com.likhit.polis.ui.q_a.QandAFragment
import com.likhit.polis.ui.recommendations.RecommendationsFragment
import com.likhit.polis.ui.userdetails.UserDetailsFragment

import java.util.ArrayList

class HomeActivity : BaseActivity(), HomeFragementListener {

    private var answers: List<String>? = ArrayList()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)

        setupToolbar(getString(R.string.app_name), true)

        replaceFragment(HomeFragment.newInstance(), HomeFragment.TAG, false)
    }

    override fun launchHome() {
        replaceFragment(HomeFragment.newInstance(), HomeFragment.TAG, true)
    }

    override fun launchUserDetails() {
        replaceFragment(UserDetailsFragment.newInstance(), UserDetailsFragment.TAG, true)
    }

    override fun launchQA() {
        replaceFragment(QandAFragment.newInstance(), QandAFragment.TAG, true)
    }

    override fun launchRecommendations(isPolicies: Boolean) {
        if (answers != null && !answers!!.isEmpty() && !isPolicies) {
            val ans = arrayOf(answers!![0], answers!![1], answers!![2], answers!![3], answers!![4])
            replaceFragment(RecommendationsFragment.newInstance(ans, isPolicies), RecommendationsFragment.TAG, true)
        } else {
            val ans: Array<String> = arrayOf("Sample")
            replaceFragment(RecommendationsFragment.newInstance(ans, isPolicies), RecommendationsFragment.TAG, true)
        }
    }

    override fun launchPolicyDetails(policy: Policy) {
        replaceFragment(DetailPolicyFragment.newInstance(policy), DetailPolicyFragment.TAG, true)
    }

    fun updateAnswers(ansers: List<String>) {

        if (answers == null) {
            answers = ArrayList()
        }
        answers = ansers
    }

    companion object {

        private val TAG = "HomeActivity"
    }
}
